lib/config/genconfig/hadeployconfig.go

```diff
@@ -0,0 +1,1102 @@
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptHaveCustomCerts(nodeType string) (customCerts, customCertsPerNode bool, err error) {
+	if c.HasCustomCerts {
+		customCerts, err = c.Prompt.Confirm("Do you have custom certs for "+nodeType+" Nodes", "yes", "no")
+		if err != nil {
+			return
+		}
+
+		if customCerts {
+			commonCerts, err1 := c.Prompt.Confirm("Do each "+nodeType+" Node have same Certs", "yes", "no")
+			if err1 != nil {
+				return commonCerts, customCertsPerNode, err1
+			}
+			customCertsPerNode = !commonCerts
+		}
+	}
+	return
+}
+func (c *HaDeployConfigGen) PromptCert(ip, nodeType, certName string) (certVal string, err error) {
+	msgCert := "Provide " + certName + " file path for " + nodeType
+	if ip != "" {
+		msgCert += " Node on IP: " + ip
+	} else {
+		msgCert += " Service"
+	}
+	certFilePath, err := c.Prompt.InputExistingFilePath(msgCert)
+	if err != nil {
+		return
+	}
+	certFile, err := c.FileUtils.ReadFile(certFilePath)
+	if err != nil {
+		return
+	}
+	certVal = string(certFile)
+	certVal = strings.TrimSpace(certVal)
+	return
+}
+
+func (c *HaDeployConfigGen) PromptAutomateNodes() (err error) {
+	automateNodeCount, err := c.Prompt.InputIntRange("No. of Automate Nodes", 1, 100)
+	if err != nil {
+		return
+	}
+	c.Config.InitAutomate().InitConfig()
+	c.Config.Automate.Config.InstanceCount = fmt.Sprint(automateNodeCount)
+
+	hasCustomCerts, hasCustomCertsPerNode, err := c.PromptHaveCustomCerts(AUTOMATE_NODETYPE)
+	if err != nil {
+		return err
+	}
+
+	c.Config.Automate.Config.EnableCustomCerts = hasCustomCerts
+
+	err = c.PromptAutomateCerts(hasCustomCerts, hasCustomCertsPerNode)
+	if err != nil {
+		return err
+	}
+
+	c.Config.InitExistingInfra().InitConfig()
+	for i := 0; i < automateNodeCount; i++ {
+		err = c.PromptAutomateIpCerts(i, hasCustomCerts, hasCustomCertsPerNode)
+		if err != nil {
+			return err
+		}
+	}
+	return
+}
+
+func (c HaDeployConfigGen) PromptAutomateIpCerts(i int, hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	ip, err1 := c.PromptNodeIp(fmt.Sprintf(PROMPT_NODE_IP, i+1))
+	if err1 != nil {
+		return err1
+	}
+	c.Config.ExistingInfra.Config.AutomatePrivateIps = append(c.Config.ExistingInfra.Config.AutomatePrivateIps, ip)
+	if hasCustomCerts && hasCustomCertsPerNode {
+		priKey, err := c.PromptCert(ip, AUTOMATE_NODETYPE, PRI_KEY)
+		if err != nil {
+			return err
+		}
+
+		pubCert, err := c.PromptCert(ip, AUTOMATE_NODETYPE, PUB_CERT)
+		if err != nil {
+			return err
+		}
+
+		nodeCert := config.CertByIP{
+			IP:         ip,
+			PrivateKey: priKey,
+			PublicKey:  pubCert,
+		}
+		*c.Config.Automate.Config.CertsByIP = append(*c.Config.Automate.Config.CertsByIP, nodeCert)
+	}
+	return
+}
+
+func (c HaDeployConfigGen) PromptAutomateCerts(hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	if hasCustomCerts && !hasCustomCertsPerNode {
+		priKey, err := c.PromptCert("", AUTOMATE_NODETYPE, PRI_KEY)
+		if err != nil {
```

Comment:
in case of more than 1 node count, the IPs should be unique, we can add checks for that

Reply:
I will consider this as feature enhancement for next time.